File: examples/python/compare_versions.py
"""Compare two NEVRA strings and print which is newer."""

import argparse

from rpm_rs import Nevra

parser = argparse.ArgumentParser(description=__doc__)
parser.add_argument("nevra1", help="First NEVRA string")
parser.add_argument("nevra2", help="Second NEVRA string")
args = parser.parse_args()

a = Nevra.parse(args.nevra1)
b = Nevra.parse(args.nevra2)

if a < b:
    print(f"{a} < {b}")
elif a == b:
    print(f"{a} == {b}")
else:
    print(f"{a} > {b}")

File: examples/compare-versions.rs
use clap::Parser;
use rpm::Nevra;
use std::cmp::Ordering;

/// Compare two NEVRA strings and print which is newer.
#[derive(Parser)]
#[command(version)]
struct Args {
    /// First NEVRA (e.g. "bash-0:5.2.15-5.el9.x86_64")
    a: String,

    /// Second NEVRA (e.g. "bash-0:5.2.21-1.el9.x86_64")
    b: String,
}

fn main() {
    let args = Args::parse();

    let a = Nevra::parse(&args.a);
    let b = Nevra::parse(&args.b);

    match a.cmp(&b) {
        Ordering::Less => println!("{a} < {b}"),
        Ordering::Equal => println!("{a} == {b}"),
        Ordering::Greater => println!("{a} > {b}"),
    }
}
